include/openssl/evp.h

```diff
@@ -1729,11 +1729,11 @@ int EVP_PKEY_derive_init(EVP_PKEY_CTX *ctx);
 int EVP_PKEY_derive_set_peer(EVP_PKEY_CTX *ctx, EVP_PKEY *peer);
 int EVP_PKEY_derive(EVP_PKEY_CTX *ctx, unsigned char *key, size_t *keylen);
 
-int EVP_KEM_encapsulate_init(EVP_PKEY_CTX *ctx);
+int EVP_KEM_encapsulate_init(EVP_PKEY_CTX *ctx, const char *name);
 int EVP_KEM_encapsulate(EVP_PKEY_CTX *ctx,
                         unsigned char *wrappedkey, size_t *wrappedkeylen,
                         unsigned char *genkey, size_t *genkeylen);
-int EVP_KEM_decapsulate_init(EVP_PKEY_CTX *ctx);
+int EVP_KEM_decapsulate_init(EVP_PKEY_CTX *ctx, const char *name);
 int EVP_KEM_decapsulate(EVP_PKEY_CTX *ctx,
```

Comment:
So while all other operations done with EVP_PKEYs are commonly named `EVP_PKEY_verb_init()` / `EVP_PKEY_verb()` or  `EVP_VerbInit()` / `EVP_Verb()` (where the "verb" is whatever verb we have chosen for a particular operation), we now get a third variant.

Why not these?

``` C
int EVP_PKEY_encapsulate_init(EVP_PKEY_CTX *ctx);
int EVP_PKEY_encapsulate_init(EVP_PKEY_CTX *ctx, const char *name);
int EVP_PKEY_encapsulate(EVP_PKEY_CTX *ctx,
                        unsigned char *wrappedkey, size_t *wrappedkeylen,
                        unsigned char *genkey, size_t *genkeylen);
int EVP_PKEY_decapsulate_init(EVP_PKEY_CTX *ctx);
int EVP_PKEY_decapsulate_init(EVP_PKEY_CTX *ctx, const char *name);
int EVP_PKEY_decapsulate(EVP_PKEY_CTX *ctx,
                        unsigned char *unwrapped, size_t *unwrappedlen,
                        const unsigned char *wrapped, size_t wrappedlen);
```

If you want previous examples for this pattern:

``` console
$ grep 'int EVP_PKEY_[a-z_]*_init' include/openssl/evp.h 
int EVP_PKEY_sign_init(EVP_PKEY_CTX *ctx);
int EVP_PKEY_verify_init(EVP_PKEY_CTX *ctx);
int EVP_PKEY_verify_recover_init(EVP_PKEY_CTX *ctx);
int EVP_PKEY_encrypt_init(EVP_PKEY_CTX *ctx);
int EVP_PKEY_decrypt_init(EVP_PKEY_CTX *ctx);
int EVP_PKEY_derive_init(EVP_PKEY_CTX *ctx);
int EVP_PKEY_param_fromdata_init(EVP_PKEY_CTX *ctx);
int EVP_PKEY_key_fromdata_init(EVP_PKEY_CTX *ctx);
int EVP_PKEY_paramgen_init(EVP_PKEY_CTX *ctx);
int EVP_PKEY_keygen_init(EVP_PKEY_CTX *ctx);
```

Reply:
Yep my bad - fixed..